app/jobs/reservation_job.rb

```diff
@@ -0,0 +1,44 @@
+class ReservationJob < ActiveJob::Base
+  queue_as :default
+
+  def perform
+    if enabled
+      log_start
+      run
+      clean
+      log_completion
+    else
+      log_disabled
+    end
+  end
+
+  private
+
+  def enabled
+    true
+  end
+
+  def log_start(type: '', task: '')
+    Rails.logger.info "Found #{collection.count} #{type} reservations, #{task}"
+  end
+
+  def log_disabled(task: '')
+    Rails.logger.info "Reservations is not configured to #{task}. "\
+      'Please change the application settings if you wish to do so.'
+  end
+
+  def log_completion
+    Rails.logger.info 'Done!'
+  end
+
+  def run
```

Comment:
I'm not so sure about that, although it may just be a matter of personal preference. In my mind I'd like the default behavior of a child job to be as safe as possible, hence `enabled` defaulting to `false` which means that the creation of a child job without the appropriate setup doesn't actually do anything (and therefore can't accidentally break things). If there are jobs we always want to run we can manually override `enabled` in the child jobs, which is also where we're ensuring that we properly set up the job. On the flip side we can just default to an empty `run` method, but then there's still weird logging happening regardless (which is why my preference was to default to disabled). It's not a huge deal either way, and if you're comfortable with the base job class as-is then we can just leave it; I just wanted to explain my thought process on requiring `enabled` to be set to `false` by default.